### ISSUE TITLE
Strings.resx generating dead properties for EventSource entries

### ISSUE THREAD

**Issue by stephentoub** [dotnet/buildtools#449]
For when generating a manifest, EventSource supports messages for each event, with text that can be written to provide a human-readable representation of the event's data.  Because this text can then be localized, these messages can be put into the .resx, and EventSource is able to pull them from the resources.  However, it pulls them directly, not via the property accessors generated into SR.cs/vb.  As a result, any project that has an EventSource typically has a bunch of property accessors that are generated but are dead code.

We should modify the resource SR.cs/vb generator to support excluding certain resources from being code gen'd, e.g. if there's a ```<Comment/>``` that contains the text "DisableCodeGeneration".

**Comment by bartonjs** [comment]
Alternatively, would it make sense to have a separate resx for EventSource strings, which doesn't get generated into SR?  (Just thinking aloud)

**Comment by tarekgh** [comment]
CC @vancem 

we had some old discussions before to have EventSource not using localized strings at all because all of it's data just viewed by the developers and not really the end users. it was just a legacy from the desktop having EventSource to include localized resources.

**Comment by stephentoub** [comment]
> we had some old discussions before to have EventSource not using localized strings at all because all of it's data just viewed by the developers and not really the end users

Is that the desired outcome?  Most if not all of the EventSources in corefx are currently using localized mesages via Strings.resx, and I don't see any issues filed to fix them up.

**Comment by tarekgh** [comment]
@stephentoub it was a mail thread discussion. I'll forward it to you.

**Comment by stephentoub** [comment]
Ok.  I'm fine with moving in that direction if that's the desired outcome.  The top-level issue I wanted to raise here is that we're injecting dead code into the assemblies.

**Comment by davidsh** [comment]
I don't think it is the desired outcome.  There are merits to having the strings be in a RESX file and eventually having the ability to localize them.

**Comment by tarekgh** [comment]
ok, will assume we'll continue having a localized strings. I'll fix the build tools to allow marking resx entries to not generate a code for and then will assign the issue to @davidsh to clean up the resources and mark the resource entries accordingly.

**Comment by stephentoub** [comment]
Would @bartonjs' suggestion work?  I'm also ok with separating the relevant resources out into a separate .resx.

**Comment by tarekgh** [comment]
I am not preferring @bartonjs idea because in the VS resources targets we have some process to detect the framework libraries and avoid them to go through some specific build tasks. the detection is done through the pattern of the resources in the library. if we are going to have another .resx in the library, that can be not get merged in the UWP apps except if we are willing to change VS targets files which is not trivial to do.

**Comment by ViktorHofer** [comment]
// Auto-generated message

Going forward, the .NET team is using https://github.com/dotnet/arcade to develop the code and issues formerly in this repository. Feel free to reopen/move this issue if it still applies to servicing branches in this repository, or source code which now resides in the Arcade repository.
